proposals/nnnn-move-function.md

```diff
@@ -0,0 +1,455 @@
+  var y = x
+  longAlgorithmUsing(&y)
+  // We think we no longer use y after this point...
+  consumeFinalY(move(y))
+
+  // ...and x will be unique again here...
+  x.append(7)
+
+  // ...but this additional use of y snuck in:
+  useYAgain(y)
+}
+```
+
+In this case, we get the following output from the compiler as expected:
+
+```
+test.swift:10:7: error: 'y' used after being moved
+  var y = x
+      ^
+test.swift:13:17: note: move here
+  consumeFinalY(move(y))
+                ^
+test.swift:19:13: note: use here
+  useYAgain(y)
+            ^
+```
+
+Note how the compiler gives all of the information that we need to resolve
+this: it says where the move was and where the later uses that
+cause the problem are, alerting the programmer to what parts of the code are
+trying to extend the lifetime of the value beyond our expected endpoint.
+
+Note that `move` only ends the lifetime of a specific local variable binding.
+It is not tied to the lifetime of the underlying value or any particular
+object instance. If we assign another binding `other` to the same value of `x`,
+we can use that other binding after we end the lifetime of `x`, as in:
+
+```
+func useX(_ x: SomeClassType) -> () {}
+func consumeX(_ x: __owned SomeClassType) -> () {}
+
+func f() -> () {
+  let x = ...
+  useX(x)
+  let other = x   // other is a new binding used to extend the lifetime of x
+  _ = move(x) // x's lifetime ends
+  useX(other)     // other is used here... no problem.
+  consumeX(other) // other is used here... no problem.
+}
+```
+
+In fact, each variable's lifetime is tracked independently, and gets a separate
+diagnostic if used after move. If we try to compile this:
+
+```
+func useX(_ x: SomeClassType) -> () {}
+func consumeX(_ x: __owned SomeClassType) -> () {}
+
+func f() -> () {
+  let x = ...
+  useX(x)
+  let other = x
+  let _ = move(x)
+  useX(move(other))
+  consumeX(other)
+  useX(x)
+}
+```
+
+we get separate diagnostics for each variable:
+
+```
+test.swift:7:15: error: 'x' used after being moved
+  let x = ...
+          ^
+test.swift:10:11: note: move here
+  let _ = move(x)
+          ^
+test.swift:13:3: note: use here
+  useX(x)
+  ^
+test.swift:9:7: error: 'other' used after being moved
+  let other = x
+      ^
+test.swift:11:8: note: move here
+  useX(move(other))
+       ^
+test.swift:12:3: note: use here
+  consumeX(other)
+  ^
+```
+
+If one applies move to a local `var`, then a new value can be assigned into
+the variable after an old value has been moved out. One can
+begin using the var again after one re-assigns to the var:
+
+```
+func f() {
+  var x = getValue()
+  let _ = move(x)
+  useX(x) // error: no value in x
+  x = getValue()
+  useX(x) // ok, x has a new value here
+}
+```
+
+This follows from move being applied to the binding (`x`), not the value in the
+binding (the value returned from `getValue()`).
+
+We also support applying the move operation to consuming function arguments:
```

Comment:
"We also support" is a statement about your implementation.  Rather,  "The move operation can also be applied to ...." is a statement about the language feature.